### PR TITLE
Clean up pre-commit

### PR DESCRIPTION
**Pull request by blink1073**
- Move some checks to CI-only
- Update docs
- Add doc8 check

### CODE REVIEW

File: app/package.json [hunk 1]
```diff
@@ -21,62 +21,62 @@
...
-    "@jupyterlab/theme-light-extension": "~4.0.0-alpha.8",
-    "@jupyterlab/tooltip": "~4.0.0-alpha.8",
-    "@jupyterlab/tooltip-extension": "~4.0.0-alpha.8",
-    "@jupyterlab/translation": "~4.0.0-alpha.8",
-    "@jupyterlab/translation-extension": "~4.0.0-alpha.8",
-    "@jupyterlab/ui-components": "~4.0.0-alpha.23",
-    "@jupyterlab/user": "~4.0.0-alpha.8",
-    "@jupyterlab/user-extension": "~4.0.0-alpha.8",
-    "@jupyterlab/vega5-extension": "~4.0.0-alpha.8",
+    "@jupyterlab/application": "~4.0.0-alpha.7",
```
Comment (jtpio) [inline app/package.json:24]: Looks like these changes were unintentional?

Comment (blink1073) [reply app/package.json:24]: They came from the integrity check that ran on push

Comment (jtpio) [reply app/package.json:24]: Right the check is pretty simple for now and picks the versions of the locally installed packages (need to fix that).

Comment (blink1073) [reply app/package.json]: Gotcha, reverted after a git clean.

**Review by jtpio — APPROVED** [review]
Thanks!

File: pyproject.toml [hunk 2]
```diff
@@ -29,7 +29,5 @@ timeout = 300
 # timeout_method = "thread"
 filterwarnings = [
     "error",
-    # Pending release of https://github.com/jupyterlab/jupyterlab_server/pull/259
-    "ignore:metadata:DeprecationWarning",
```
[the remark this answers is not in the document]
Comment (blink1073) [reply pyproject.toml:33]: Done: https://github.com/jupyterlab/jupyterlab_server/releases/tag/v2.13.0

[the remark this answers is not in the document]
Comment (jtpio) [reply pyproject.toml:33]: Thanks!

**Pull request merged**